perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -339,27 +345,44 @@ def _ProcessMultiStreamResults(raw_result, operation, results, metadata=None):
 
   records = json.loads(raw_result)
   metric_name = 'Multi-stream %s latency' % operation
-  # TODO: when object size distributions are added, support multiple
-  # sizes.
-  object_size = records[0]['size']
 
-  logging.info('Processing %s multi-stream %s results for object size %s',
-               len(records), operation, object_size)
-
-  # Once we land size distributions, we will report different latency
-  # for each object size.
   metadata = metadata.copy()
-  metadata['object_size_B'] = object_size
   metadata['num_streams'] = FLAGS.object_storage_multistream_num_streams
   metadata['objects_per_stream'] = (
       FLAGS.object_storage_multistream_objects_per_stream)
 
+  overall_metadata = metadata.copy()
+  # I would love to publish the full distribution in the metadata, but
+  # I'm afraid that including a complex value like a dictionary in
+  # metadata that is otherwise simple Python objects would break
+  # parsers. Also, we'd need to implement a custom serializer unless
+  # we wanted regular expression parsing of metadata to depend on
+  # Python's hash function. I add 'object_size_B':'distribution' for
+  # the full results because the metadata for the per-object-size
+  # values will include the 'object_size_B' field, and searching can
+  # be easier when all the records you care about have the same
+  # metadata fields.
+  overall_metadata['object_size_B'] = 'distribution'
+  logging.info('Processing %s multi-stream %s results for the full '
+               'distribution.', len(records), operation)
   _AppendPercentilesToResults(
       results,
       (record['latency'] for record in records),
       metric_name,
       'sec',
-      metadata)
+      overall_metadata)
+
+  for size in sizes:
+    metadata = metadata.copy()
```

Comment:
I don't think that will work, because each of the Sample object stores a pointer to its metadata dictionary. If I made only one copy, then the metadata for all of the samples would change when I updated inside the loop.